src/core/symbology/qgsrendererrange.cpp

```diff
@@ -163,13 +163,13 @@ const int QgsRendererRangeLabelFormat::MIN_PRECISION = -6;
 QgsRendererRangeLabelFormat::QgsRendererRangeLabelFormat()
   : mFormat( QStringLiteral( "%1 - %2" ) )
   , mReTrailingZeroes( "[.,]?0*$" )
-  , mReNegativeZero( "^\\-0(?:[.,]0*)?$" )
+  , mReNegativeZero( QRegularExpression::anchoredPattern( "^\\-0(?:[.,]0*)?$" ) )
```

Comment:
could be moved to header to avoid the duplicate definition (or use delegated constructor)